### PR TITLE
Add Vue 3 support with vue-demi

### LINKED ISSUE #46: Vue 3 Support
**TLDR**: We're looking to use vue-ld in a Vue 3 application and it doesn't look like there's support for Vue 3 yet. Is there a timeline on support?

In a Vue 3 app we're configuring `main.js` like this:
```
import { createApp } from 'vue'
import App from './App.vue'
import { VueLd } from 'vue-ld';

const app = createApp(App)
app.use(VueLd, {
    clientSideId: 'YOUR_CLIENT_SIDE_ID',
    user: {
        key: '...',
        email: '...',
        name: '...',
    },
    options: {
        // Standard LaunchDarkly JavaScript SDK options
    },
});
app.mount('#app')
```

When we run the application we're finding that the page doesn't load and we're getting the following error in the console:
```
Uncaught TypeError: Vue.observable is not a function
    at Object.install (index.es.js:1010)
    at Object.use (runtime-core.esm-bundler.js:3003)
    at main.js:6
```

I believe this could be because `Vue.observable` is not supported in Vue 3. Is there a timeline for when vue-ld will support Vue 3? Or, can you point me at a similar tool for Launch Darkly that supports Vue 3?

Thank you!

### PR DESCRIPTION
**Pull request by james-black**
Story: https://app.shortcut.com/dashhudson/story/54579

Todo:
- [x]  Make units test work in Vue 3
- [x]  Fix rendering regression
- [x] Update GitHub Actions to reflect changes to the package.json scripts

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -73,7 +74,12 @@ export default {
     } else {
       $ld = initialize({ clientSideId, user, ldOptions, readyBeforeIdentify });
     }
-    // eslint-disable-next-line no-param-reassign
-    Vue.prototype.$ld = Vue.observable($ld);
+
+    if (isVue2) {
+      // eslint-disable-next-line no-param-reassign
+      vue.prototype.$ld = vue.observable($ld);
```
Comment (benjamin-t-frost) [inline src/plugin.js]: By the docs, it seems `reactive` should point to observable or ref depending on the environment; is there a reason to use both?

Comment (james-black) [reply src/plugin.js]: Whoops, yeah there isn't, will fix that

File: src/plugin.js [hunk 2]
```diff
@@ -73,7 +74,12 @@ export default {
...
       $ld = initialize({ clientSideId, user, ldOptions, readyBeforeIdentify });
     }
-    // eslint-disable-next-line no-param-reassign
-    Vue.prototype.$ld = Vue.observable($ld);
+
+    if (isVue2) {
+      // eslint-disable-next-line no-param-reassign
+      vue.prototype.$ld = $ld;
+    } else {
+      vue.provide('$ld', $ld);
```
Comment (benjamin-t-frost) [inline src/plugin.js]: Why chose to use provide over assigning it through `app.config.globalProperties`?

https://v3.vuejs.org/guide/plugins.html#writing-a-plugin

File: tests/unit/ld-redirect.spec.js [hunk 3]
```diff
@@ -18,39 +17,38 @@ const createMixin = (props) => {
...
+    server.autoRespond = true;
+    server.autoRespondAfter = 0;
+  });
+
   afterEach(() => {
     server.restore();
   });
 
-  it('redirects without feature flag', async () => {
+  it('redirects wiout feature flag', async () => {
```
Comment (benjamin-t-frost) [inline tests/unit/ld-redirect.spec.js]: without ~

File: tests/unit/utils.js [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+    });
+    wrapper = mount(component, {
+      localVue,
+      mixins,
+      mocks,
+      propsData: props,
+    });
+  } else {
+    // eslint-disable-next-line no-shadow
+    const { mount } = await import('@vue/test-utils');
```
Comment (benjamin-t-frost) [inline tests/unit/utils.js]: We probably do want to follow the no-shadow rule. You can rename imports like this:
```javascript
const { mount: originalMount } = await import('@vue/test-utils');
```

File: src/plugin.js (hunk 2 above)
Comment (james-black) [reply src/plugin.js]: I was having trouble getting components to react to changes with a reactive globalProperty, but revisiting it now it looks like it does work as well so I'll definitely switch back to this.

**Review by benjamin-t-frost — APPROVED** [review]
Thanks for getting through this!

**Review by glounder — APPROVED** [review]

**Pull request merged**